### PR TITLE
feat: Add endpoint for upcoming events

### LINKED ISSUE #6955: Filter upcoming events on the basis of their end date and time.
**Describe the bug**

Currently the events are being filtered on the basis of end date only. If an event is single day event of 2hrs, then the vent will get removed from upcoming events on the next day. The event should not be shown in upcoming events if it is over.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create an event.
2. Set event start time and end time for a small period.
3. After the event is over, it is still shown in upcoming events.

**Expected behaviour**

Once the event is over, it should not be shown in the upcoming events or browse events.

**Screenshots**

Screenshot from 2019-04-15 18-05-18

Screenshot from 2019-04-15 18-05-11

**Additional context**

I am working on this issue.

### PR DESCRIPTION
**Pull request by mansiag**
Fixes #6955 

#### Short description of what this resolves:
The filtering for upcoming events is done on the frontend using the filter query parameter. This should be handled by the server, so an endpoint /events/upcoming is created to automatically return the upcoming events.

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/events.py [hunk 1]
```diff
@@ -857,3 +857,45 @@ def clear_export_urls(event):
...
+        :return:
+        """
+        if 'Authorization' in request.headers and (
+            has_access('is_admin') or kwargs.get('user_id')
+        ):
+            self.schema = EventSchema
+        else:
+            self.schema = EventSchemaPublic
+
+        self.schema.self_view_many = 'v1.upcoming_event_list'
```
Comment (iamareebjamal) [inline app/api/events.py:881]: Isn't this already defined in superclass?

File: app/api/events.py [hunk 2]
```diff
@@ -857,3 +857,45 @@ def clear_export_urls(event):
...
+            Event.state=='published',
+            Event.privacy=='public').order_by(Event.starts_at)
+        return query_
+
+    schema = EventSchema
+    data_layer = {
+        'session': db.session,
+        'model': Event,
+        'methods': {'query': query},
+    }
```
Comment (iamareebjamal) [inline app/api/events.py:901]: This is already defined in superclass as well I believe

**Comment by iamareebjamal** [comment]
Please also add a dredd test in the docs to ensure it works and conforms to the docs

File: app/api/events.py (hunk 2 above)
Comment (mansiag) [reply app/api/events.py:901]: The query method is not invoked for returning the desired events without defining data_layer again in the child class. It should take methods in data_layer variable from superclass, but this isn't happening.

Comment (iamareebjamal) [reply app/api/events.py:901]: Hmm. The library is not really well maintained. OK

Comment (iamareebjamal) [reply app/api/events.py:901]: Check other things. If they are working or not

**Comment by mansiag** [comment]
> Please also add a dredd test in the docs to ensure it works and conforms to the docs

I am quite confused if the URL parameters filter and sort should still be added to the docs, as one may want to further filter these events ? 
Also, sort parameter will be of no use, as we are performing sort in the server code, should sorting be removed from the server and moved to the frontend?

File: app/api/events.py (hunk 2 above)
Comment (mansiag) [reply app/api/events.py:901]: Yeah, other things are working fine.

**Comment by iamareebjamal** [comment]
You just need to add the URL without filter and sorting

**Comment by iamareebjamal** [comment]
Actual response returned no events https://app.apiary.io/public/tests/run/acb5ac2f-cba0-44d1-bd68-48248bc5ede0

**Comment by iamareebjamal** [comment]
Also, remove the page size and number

File: app/api/events.py [hunk 3]
```diff
@@ -857,3 +857,38 @@ def clear_export_urls(event):
...
+        self.schema.self_view_many = 'v1.upcoming_event_list'
+
+    def query(self, view_kwargs):
+        """
+        query method for upcoming events list
+        :param view_kwargs:
+        :return:
+        """
+        current_time = datetime.now(pytz.utc)
+        query_ = self.session.query(Event).filter(
```
Comment (iamareebjamal) [inline app/api/events.py:884]: This does not filter deleted events. Maybe that is done automatically in the library, but can you please still verify that deleted events are not being returned?

Comment (mansiag) [reply app/api/events.py:884]: Yes, deleted events are not returned.

**Comment by mansiag** [comment]
Don't know why test hasn't run https://app.apiary.io/public/tests/run/245ae4a9-6419-48ef-bad2-20e669787022

**Comment by iamareebjamal** [comment]
Working fine. Can you please change PR title to match semantic guidelines?

**Pull request merged**